wazo_webhookd/services/mobile/plugin.py

```diff
@@ -0,0 +1,213 @@
+
+
+class PushNotification(object):
+
+    def __init__(self, external_token, apns_token, external_config):
+        self.token = external_token
+        self.apns_token = apns_token
+        self.external_config = external_config
+
+    def send_notification(self, data):
+        message_title = None
+        message_body = None
+
+        is_incoming_call = data.get('notification_type') == 'incomingCall'
+        is_voicemail = data.get('notification_type') == 'voicemailReceived'
+        is_message = data.get('notification_type') == 'messageReceived'
+
+        if is_incoming_call:
+            message_title = 'Incoming Call'
+            message_body = 'From: {}'.format(data['items']['peer_caller_id_number'])
+            channel_id = 'wazo-notification-call'
+
+        if is_voicemail:
+            message_title = 'New voicemail'
+            message_body = 'From: {}'.format(data['items']['message']['caller_id_num'])
+            channel_id = 'wazo-notification-voicemail'
+
+        if is_message:
+            message_title = data['items']['alias']
+            message_body = data['items']['content']
+            channel_id = 'wazo-notification-chat'
+
+        if self.apns_token and channel_id == 'wazo-notification-call':
+            try:
+                return self._send_via_apn(self.apns_token, data)
+            except (apns2_errors.ServiceUnavailable,
+                    apns2_errors.InternalServerError) as e:
+                raise HookRetry({"error": str(e)})
+        else:
+            try:
+                return self._send_via_fcm(message_title, message_body, channel_id, data)
+            except RetryAfterException as e:
+                raise HookRetry({"error": str(e)})
+
+    def _send_via_fcm(self, message_title, message_body, channel_id, data):
+        push_service = FCMNotification(api_key=self.external_config['fcm_api_key'])
+
+        if (message_title and message_body):
+
+            if channel_id == 'wazo-notification-call':
+                notification = push_service.notify_single_device(
+                    registration_id=self.token,
+                    data_message=data,
+                    extra_notification_kwargs=dict(priority='high'))
+            else:
+                notification = push_service.notify_single_device(
+                    registration_id=self.token,
+                    message_title=message_title,
+                    message_body=message_body,
+                    badge=1,
+                    extra_notification_kwargs=dict(android_channel_id=channel_id),
+                    data_message=data)
+
+            if notification.get('failure') != 0:
+                logger.error('Error to send push notification: %s', notification)
+            return notification
+
+    def _send_via_apn(self, apns_token, data):
+        payload = Payload(alert=data, sound="default", badge=1)
+        client = APNsClient(self.external_config['ios_apns_cert'],
```

Comment:
I don't know what could happen here but the content of `/tmp/ios.pem` is the last one that was written to the file. Which could be for another tenant.